substrate/client/api/src/execution_extensions.rs

```diff
@@ -116,17 +116,20 @@ impl<Block: BlockT> ExecutionExtensions<Block> {
 		*self.extensions_factory.write() = Box::new(maker);
 	}
 
-	/// Based on the execution context and capabilities it produces
-	/// the extensions object to support desired set of APIs.
+	/// Produces default extensions based on the input parameters.
 	pub fn extensions(
 		&self,
 		block_hash: Block::Hash,
 		block_number: NumberFor<Block>,
+		proof_recorder: Option<&Recorder<HashingFor<Block>>>,
```

Comment:
do we need to pass recorder down to this function?
why cannot we register recorder after calling this function - we'll have extensions available so we could just push something there.

Reply:
You mean we could add the extension [here](https://github.com/skunert/polkadot-sdk/blob/795e68e4379caaf64bec3a0c7fe3207c47dd2d28/substrate/client/service/src/client/client.rs#L1761)?

Yeah that would certainly be possible. While not strictly necessary to do this here, I was just under the impression that this `ExecutionExtensions` struct should be responsible for registering extensions based on input parameters.

Reply:
I agree that it is not super nice to pass the recorder here. However I think the approach has some benefits:
- All default extensions initialized in one place
- You can use the client but still customize the default extensions that are registered by passing a different `ExecutionExtensions` object

> Just use mut extensions argument to pass what is needed.

I would still need to pass the recorder from this runtime api instance to the client to register it on the extensions struct.

Reply:
> All default extensions initialized in one place

recorder one seems to be optional,

> I would still need to pass the recorder from this runtime api instance to the client to register it on the extensions struct.

Not sure what you mean here? Is this about this: https://github.com/paritytech/polkadot-sdk/blob/3701a7ab664a2d22d86401f5b2bb18a2f26e29a5/substrate/client/api/src/execution_extensions.rs#L131


My idea was to simply inject recorder extension into `self.extensions` here:
https://github.com/paritytech/polkadot-sdk/blob/7cbe0c76ef8fd2aabf9f07de0156941ce3ed44b0/substrate/primitives/api/proc-macro/src/impl_runtime_apis.rs#L583-L602